frontend/amundsen_application/static/js/utils/ownerUtils.ts

```diff
@@ -0,0 +1,65 @@
+import { OwnerDict, ResourceType, UpdateOwnerPayload, User } from 'interfaces';
+import { indexUsersEnabled } from 'config/config-utils';
+import { OwnerItemProps } from 'components/OwnerEditor';
+import * as API from 'ducks/tableMetadata/api/v0';
+
+/**
+ * Converts a list of Users to a dictionary of users OwnersDict
+ */
+export function getOwnersDictFromUsers(owners: User[]): OwnerDict {
+  const ownerObj = owners.reduce((resultObj, currentOwner) => {
+    resultObj[currentOwner.user_id] = currentOwner as User;
+    return resultObj;
+  }, {});
+  return ownerObj;
+}
+
+/**
+ * Converts a list of Users to OwnerItemProps
+ */
+export function getOwnerItemPropsFromUsers(
+  owners: User[],
+  profileLinkSource: string
+): OwnerItemProps {
+  return owners.reduce((obj, user) => {
+    const { profile_url, user_id, display_name } = user;
+    let profileLink = profile_url;
+    let isExternalLink = true;
+    if (indexUsersEnabled()) {
+      isExternalLink = false;
+      profileLink = `/user/${user_id}?source=${profileLinkSource}`;
+    }
+    obj[user_id] = {
+      label: display_name,
+      link: profileLink,
+      isExternal: isExternalLink,
+    };
+    return obj;
```

Comment:
```suggestion
    return {
        ...obj,
        [user_id] = {
          label: display_name,
          link: profileLink,
          isExternal: isExternalLink,
       }
    }
;```

Reply:
Wouldn't the spread operator be redundant since this is inside a reduce function?

Reply:
You mean the creation of a new object is not necessary?
It is a good practice though, in most cases we prefer to create new objects than mutate current objects.